simplecpp.cpp

```diff
@@ -2681,6 +2683,54 @@ static bool isCpp17OrLater(const simplecpp::DUI &dui)
     return !std_ver.empty() && (std_ver >= "201703L");
 }
 
+
+static std::string currentDirectoryOSCalc() {
+#ifdef SIMPLECPP_WINDOWS
+    TCHAR NPath[MAX_PATH];
+    GetCurrentDirectory(MAX_PATH, NPath);
+#ifdef _UNICODE
+    // convert the result from TCHAR* to char*
+    char NPathA[MAX_PATH];
+    ::WideCharToMultiByte(CP_ACP, 0, NPath, lstrlen(NPath), NPathA, MAX_PATH, NULL, NULL);
+    return NPathA;
+#else
+    // in this case, TCHAR* is just defined to be a char*
+    return NPath;
+#endif
+#else
+    const std::size_t size = 1024;
+    char the_path[size];
+    getcwd(the_path, size);
+    return the_path;
+#endif
+}
+
+static const std::string& currentDirectory() {
+    static const std::string curdir = simplecpp::simplifyPath(currentDirectoryOSCalc());
+    return curdir;
+}
+
+static std::string toAbsolutePath(const std::string& path) {
+    if (path.empty()) {
+        return path;// preserve error file path that is indicated by an empty string
+    }
+    if (!isAbsolutePath(path)) {
+        return currentDirectory() + "/" + path;
```

Comment:
should we use simplifyPath here also maybe?  imagine that "path" is ".."?

Reply:
Done